pkg/internal/testutils/generate_pod_yamls.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package testutils
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/np-guard/netpol-analyzer/pkg/internal/projectpath"
+)
+
+// a test util for eval command-line; generating a new temporary directory with pod files to be used for testing.
+// eval command supports only test directories with Pod objects.
+// some of our testing directories are containing other Workload types in the manifests; in order to be able to test
+// eval command on those dirs (in unit-testing); this file funcs are used.
+// we copy the test-die into a new temporary dir, and generate into the tempDir :
+// Pod yaml files for given src and dst peers from their workload resources
+
+const (
+	tmpPattern = "temp-*"
+	exeMode    = 0o777
+)
+
+var TmpDir = filepath.Join(projectpath.Root, "temp") // cleaned up after the test is done
+
+// GenerateTempDirWithPods generates new temporary dir that copies origDir and adds yaml files of Pod kind
+// matching the input workload resources of the src and dst
+// the function returns the path of the generated temp dir.
+func GenerateTempDirWithPods(origDir, srcName, srcNs, dstName, dstNs string) (string, error) {
+	// create the TmpDir path if does not exist
+	if _, err := os.Stat(TmpDir); os.IsNotExist(err) {
+		osErr := os.Mkdir(TmpDir, exeMode)
+		if osErr != nil {
+			return "", osErr
+		}
+	}
+	// making temp directory in the temp path
```

Comment:
why is this required? isn't it sufficient to just use `TmpDir` ?

Reply:
true, done